src/test/ui/consts/const_discriminant.rs

```diff
@@ -25,7 +24,7 @@ enum SingleVariant {
 
 const TEST_V: Discriminant<SingleVariant> = discriminant(&SingleVariant::V);
 
-pub const TEST_VOID: () = {
+pub const TEST_VOID: () = { //~ evaluation of constant value failed
     // This is UB, but CTFE does not check validity so it does not detect this.
     // This is a regression test for https://github.com/rust-lang/rust/issues/89765.
```

Comment:
this comment is now outdated.

cc @rust-lang/wg-const-eval for slight changes in const eval semantics

Reply:
Does this still behave as before if we instead `transmute` from `&()` to `&Void`? Right now this doesn't really test any more what it should test.

Reply:
Also if typeck properly respects visibility, the old behavior can probably be restored by moving `Void` into another module or so? That seems better (maybe also for a bunch of the other tests)

Reply:
Makes sense. I guess I should add a test for when the type is visibly uninhabited then.

Reply:
Well this particular test was intended to cover the case where the type is uninhabited but the type checker doesn't see that. So it wouldn't make much sense to use visibly uninhabited types.